lib/rules/no-wait-for-side-effects.ts

```diff
@@ -26,6 +26,28 @@ export default createTestingLibraryRule<Options, MessageIds>({
   },
   defaultOptions: [],
   create: function (context, _, helpers) {
+    function isCallerWaitFor(
+      node: TSESTree.BlockStatement | TSESTree.CallExpression
+    ) {
```

Comment:
Can you specify here that this function returns a `boolean` please?

Reply:
shouldn't the compiler be smart enough to infer that itself? I usually tend to avoid adding the return types for inline functions.

Reply:
I find that useful when you are not really sure what type you are returning (like a React component), but here are pretty sure we want to return a boolean, so specifying the type would prevent us from returning something else. I don't know what you think, and I've probably missed few like this in the past myself 😅

Reply:
I'm open to either option 🤷

Reply:
I added explicit return type in 8e6044a. I think that's useful if you want to make sure the API does not change in the long term